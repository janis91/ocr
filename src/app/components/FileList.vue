<template>
  <div>
    <ul class="ocr-list">
      <li v-for="file in files.slice(0,5)" :key="file.id" class="ocr-list-item">{{file.name}}</li>
      <li v-if="files.length >= 6" class="ocr-list-item">...</li>
    </ul>
  </div>
</template>

<script lang="ts">
import Vue from 'vue'

export default Vue.extend({
  name: 'FileList',
  props: {
    files: {
      type: Array as () => OCAFile[],
      required: true
    }
  }
})
</script>

<style scoped>
.ocr-list {
  display: flex;
  flex-direction: column;
  padding-left: 0;
  margin-bottom: 0;
  padding-bottom: 10px;
}
.ocr-list-item:last-child {
  border-bottom: 0;
}
.ocr-list-item {
  position: relative;
  display: block;
  padding: .75rem 1.25rem;
  margin-bottom: -1px;
  border-bottom: 1px solid var(--color-border-dark);
}
</style>
