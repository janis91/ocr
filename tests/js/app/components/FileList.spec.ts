import { mount, Wrapper } from '@vue/test-utils'
import FileList from '@a/components/FileList.vue'
import { FileFixtures } from '../../fixtures/FileFixtures'

describe('FileList.vue', () => {
  let wrapper: Wrapper<any>

  beforeEach(() => {
    wrapper = mount(FileList, {
      propsData: {
        files: []
      }
    })
  })

  it('GIVEN standard props, WHEN FileList is mounted, THEN matches snapshot with zero list items.', async () => {
    expect(wrapper.contains('li')).toBeFalsy()
    expect(wrapper.element).toMatchSnapshot()
  })

  it('GIVEN one file in list, WHEN FileList is mounted, THEN matches snapshot with one list item.', async () => {
    wrapper.setProps({ files: [FileFixtures.PNG] })

    await wrapper.vm.$nextTick()

    expect(wrapper.contains('li')).toBeTruthy()
    expect(wrapper.text()).toContain(FileFixtures.PNG.name)
    expect(wrapper.element).toMatchSnapshot()
  })

  it('GIVEN five files in list, WHEN FileList is mounted, THEN matches snapshot with five list items.', async () => {
    wrapper.setProps({ files: FileFixtures.ALLOWED })

    await wrapper.vm.$nextTick()

    expect(wrapper.contains('li')).toBeTruthy()
    FileFixtures.ALLOWED.forEach(({ name }) => expect(wrapper.text()).toContain(name))
    expect(wrapper.element).toMatchSnapshot()
  })

  it('GIVEN six files in list, WHEN FileList is mounted, THEN matches snapshot with five list items plus "..." as last item.', async () => {
    wrapper.setProps({ files: [...FileFixtures.ALLOWED, FileFixtures.ADDITIONAL_PNG] })

    await wrapper.vm.$nextTick()

    expect(wrapper.contains('li')).toBeTruthy()
    FileFixtures.ALLOWED.forEach(({ name }) => expect(wrapper.text()).toContain(name))
    expect(wrapper.text()).not.toContain(FileFixtures.ADDITIONAL_PNG.name)
    expect(wrapper.text()).toContain('...')
    expect(wrapper.element).toMatchSnapshot()
  })
})
