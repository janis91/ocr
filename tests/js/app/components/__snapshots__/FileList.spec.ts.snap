// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileList.vue GIVEN five files in list, WHEN FileList is mounted, THEN matches snapshot with five list items. 1`] = `
<div>
  <ul
    class="ocr-list"
  >
    <li>
      file3.png
    </li>
    <li>
      file3.jpg
    </li>
    <li>
      file3.bmp
    </li>
    <li>
      file3.tiff
    </li>
    <li>
      file3.bmpx
    </li>
     
    <!---->
  </ul>
</div>
`;

exports[`FileList.vue GIVEN one file in list, WHEN FileList is mounted, THEN matches snapshot with one list item. 1`] = `
<div>
  <ul
    class="ocr-list"
  >
    <li>
      file3.png
    </li>
     
    <!---->
  </ul>
</div>
`;

exports[`FileList.vue GIVEN six files in list, WHEN FileList is mounted, THEN matches snapshot with five list items plus "..." as last item. 1`] = `
<div>
  <ul
    class="ocr-list"
  >
    <li>
      file3.png
    </li>
    <li>
      file3.jpg
    </li>
    <li>
      file3.bmp
    </li>
    <li>
      file3.tiff
    </li>
    <li>
      file3.bmpx
    </li>
     
    <li>
      ...
    </li>
  </ul>
</div>
`;

exports[`FileList.vue GIVEN standard props, WHEN FileList is mounted, THEN matches snapshot with zero list items. 1`] = `
<div>
  <ul
    class="ocr-list"
  >
     
    <!---->
  </ul>
</div>
`;
